src/Core/Grid/Position/UpdateHandler/DoctrinePositionUpdateHandler.php

```diff
@@ -0,0 +1,131 @@
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Position\UpdateHandler;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\ConnectionException;
+use Doctrine\DBAL\Statement;
+use PrestaShop\PrestaShop\Core\Grid\Position\Exception\PositionUpdateException;
+use PrestaShop\PrestaShop\Core\Grid\Position\PositionDefinitionInterface;
+
+/**
+ * Class DoctrinePositionUpdateHandler updates the grid positions using a Doctrine
+ * Connection.
+ */
+final class DoctrinePositionUpdateHandler implements PositionUpdateHandlerInterface
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     */
+    public function __construct(
+        Connection $connection,
+        $dbPrefix
+    ) {
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCurrentPositions(PositionDefinitionInterface $positionDefinition, $parentId = null)
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->from($this->dbPrefix . $positionDefinition->getTable(), 't')
+            ->select('t.' . $positionDefinition->getIdField() . ', t.' . $positionDefinition->getPositionField())
+            ->addOrderBy('t.' . $positionDefinition->getPositionField(), 'ASC')
+        ;
+
+        if (null !== $parentId && null !== $positionDefinition->getParentIdField()) {
+            $qb
+                ->andWhere('t.' . $positionDefinition->getParentIdField() . ' = :parentId')
+                ->setParameter('parentId', $parentId)
+            ;
+        }
+
+        $positions = $qb->execute()->fetchAll();
+        $currentPositions = [];
+        foreach ($positions as $position) {
+            $positionId = $position[$positionDefinition->getIdField()];
+            $currentPositions[$positionId] = $position[$positionDefinition->getPositionField()];
+        }
+
+        return $currentPositions;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function updatePositions(PositionDefinitionInterface $positionDefinition, array $newPositions)
+    {
+        try {
```

Comment:
I love it! we should do that everywhere (and/or imagine our own abstraction of Connection class)